pkg/reconciler/pipelinerun/resources/apply.go

```diff
@@ -94,6 +94,7 @@ func ApplyReplacements(p *v1beta1.PipelineSpec, replacements map[string]string,
 	p = p.DeepCopy()
 
 	tasks := p.Tasks
+	tasks = append(tasks, p.Finally...)
```

Comment:
NIT: I guess this could be `append(p.Tasks, p.Finally...)` ?

Reply:
Done

Reply:
I have bitten myself in the foot a few times using `append()`. I know the way @pritidesai initially wrote is safe. This may also work - but it may also introduce problems that is hard to debug since it may mutate `p.Tasks`

I don't have deep enough understanding about this, there are some articles that may elaborate about this, e.g. https://appliedgo.net/slices/